### PR TITLE
DPC 1534 modify caveats

### PR DESCRIPTION
**Pull request by MrBilnon**
### Proposed Changes

Add system x data to generated macaroon, add system x data to access token claims. When exchanging signed client token for access token, parse macaroon in client token for system id and verify the macaroon by it's root key

### Change Details

* add system x data to unsigned macaroon
* add system x data to access token
* return system x data when validating
* parse macaroon for system id and verify macaroon when requesting access token

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Modified api_test to create v2 system, and sign the client token and request an access token and verified no error of finding the system id from the macaroon.

### CODE REVIEW

File: ssas/service/server.go [hunk 1]
```diff
@@ -244,12 +247,13 @@ type CommonClaims struct {
 	SystemID string `json:"sys,omitempty"`
 	// In a registration token, GroupIDs contains a list of all groups this user is authorized to manage
 	GroupIDs []string `json:"gid,omitempty"`
-	Data     string   `json:"dat,omitempty"`
+	Data     string   `json:"group_data,omitempty"`
 	Scopes   []string `json:"scp,omitempty"`
 	// deprecated
 	ACOID string `json:"aco,omitempty"`
 	// deprecated
-	UUID string `json:"id,omitempty"`
```
[the remark this answers is not in the document]
Comment (MrBilnon) [reply ssas/service/server.go:252]: I think if you have a comment then gofmt treats it as a new "block"

File: ssas/service/server.go [hunk 2]
```diff
@@ -244,12 +247,13 @@ type CommonClaims struct {
 	SystemID string `json:"sys,omitempty"`
 	// In a registration token, GroupIDs contains a list of all groups this user is authorized to manage
 	GroupIDs []string `json:"gid,omitempty"`
-	Data     string   `json:"dat,omitempty"`
+	Data     string   `json:"group_data,omitempty"`
```
Comment (dhgreene) [inline ssas/service/server.go:250]: Consumers of SSAS v1 depend on the `dat` claim.

File: ssas/service/server.go [hunk 3]
```diff
@@ -325,6 +335,46 @@ func (s *Server) VerifyClientSignedToken(tokenString string, trackingId string)
 	}
 	return jwt.ParseWithClaims(tokenString, &CommonClaims{}, keyFunc)
 }
+
+func (s *Server) GetSystemIDFromMacaroon(issuer string) (string, error) {
```
Comment (dhgreene) [inline ssas/service/server.go:339]: Do we want a single-purpose method for finding a particular 1st party caveat? Won't the code be 90% the same for finding the group ID from the macaroon?

File: ssas/testutils.go [hunk 4]
```diff
@@ -136,3 +135,32 @@ func CreateTestXData(t *testing.T, db *gorm.DB) (creds Credentials, group Group)
 
 	return
 }
+
+func CreateTestXDataV2(t *testing.T, db *gorm.DB) (creds Credentials, group Group) {
+	groupID := RandomHexID()[0:4]
+
+	group = Group{GroupID: groupID, XData: "fake x_data"}
```
Comment (dhgreene) [inline ssas/testutils.go:142]: Same as below: let's go for some valid JSON
```suggestion
	group = Group{GroupID: groupID, XData: `{"key":"value"}`

**Review by MrMorie — APPROVED** [review]
🥇

File: ssas/service/server.go (hunk 2 above)
Comment (MrBilnon) [reply ssas/service/server.go:250]: okay thought so, the ticket seemed like it was calling out to change this field

File: ssas/service/server.go (hunk 3 above)
Comment (MrBilnon) [reply ssas/service/server.go:360]: done

**Review by MrMorie — APPROVED** [review]
🆒

**Pull request merged**